### PR TITLE
blacklist support for random maint

### LINKED ISSUE #4323: no blacklist for random rooms
https://github.com/BeeStation/BeeStation-Hornet/blob/09902e8d98a0b5eb24679de26fe054e179c153a0/code/controllers/subsystem/mapping.dm#L364

### PR DESCRIPTION
**Pull request by bloons3**
## About The Pull Request

Resolves https://github.com/BeeStation/BeeStation-Hornet/issues/4323

## Changelog
:cl:
config: Added a config to blacklist rooms from Random Maintenance 
/:cl:

### CODE REVIEW

**Comment by bloons3** [comment]
blacklisted room was removed in https://github.com/BeeStation/BeeStation-Hornet/pull/1879

File: code/controllers/subsystem/mapping.dm [hunk 1]
```diff
@@ -356,11 +356,17 @@ GLOBAL_LIST_EMPTY(the_station_areas)
 	preloadRandomRoomTemplates()
 
 /datum/controller/subsystem/mapping/proc/preloadRandomRoomTemplates()
+	var/list/banned = generateMapList("[global.config.directory]/randommaintblacklist.txt")
 	for(var/item in subtypesof(/datum/map_template/random_room))
 		var/datum/map_template/random_room/room_type = item
 		if(!(initial(room_type.mappath)))
```
Comment (Archanial) [inline code/controllers/subsystem/mapping.dm:362]:
```suggestion
		var/roompath = initial(room_type.mappath)
		if(!roompath || banned.Find(roompath))
```

File: code/controllers/subsystem/mapping.dm [hunk 2]
```diff
@@ -356,11 +356,17 @@ GLOBAL_LIST_EMPTY(the_station_areas)
...
 			continue
+
 		var/datum/map_template/random_room/R = new room_type()
+
+		if(banned.Find(R.mappath))
+			continue
+
```
Comment (Archanial) [inline code/controllers/subsystem/mapping.dm:369]:
```suggestion
```

File: code/controllers/subsystem/mapping.dm [hunk 3]
```diff
@@ -356,11 +356,17 @@ GLOBAL_LIST_EMPTY(the_station_areas)
...
 /datum/controller/subsystem/mapping/proc/preloadRandomRoomTemplates()
+	var/list/banned = generateMapList("[global.config.directory]/randommaintblacklist.txt")
 	for(var/item in subtypesof(/datum/map_template/random_room))
 		var/datum/map_template/random_room/room_type = item
 		if(!(initial(room_type.mappath)))
 			continue
+
 		var/datum/map_template/random_room/R = new room_type()
+
+		if(banned.Find(R.mappath))
```
Comment (PowerfulBacon) [inline code/controllers/subsystem/mapping.dm:367]: Check the room_id not the mappath

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/mapping.dm (hunk 3 above)
Comment (bloons3) [reply code/controllers/subsystem/mapping.dm:367]: I am mirroring the space ruin/lavaland ruin blacklist, which blacklists the map file itself and not the ruin ID.

**Review by PowerfulBacon — APPROVED** [review]
Config Changes

**Comment by bloons3** [comment]
> Config Changes

Config change isn't actually a config change. This is adding a config with the same settings as the current game.

That template was commented out because it caused runtimes. In effect, it is still disabled.

**Comment by bloons3** [comment]
no desire to maintain

**Pull request closed without merging**